framework/web/UrlRule.php

```diff
@@ -213,6 +218,95 @@ public function init()
         if (!empty($this->_routeParams)) {
             $this->_routeRule = '#^' . strtr($this->route, $tr2) . '$#u';
         }
+
+        $this->initCanonizer();
+    }
+
+    /**
+     * Initializes canonizer configuration
+     * @throws \Exception
+     */
+    protected function initCanonizer() {
+        $canonize = $this->canonize;
+
+        $default = [
+            // whether to collapse multiple slashes into one
+            'collapse-slashes' => false,
+            // whether trailing slash required
+            // true: yes
+            // false: no
+            // null: leave as is
+            'trailing-slash' => null,
+            // what action to perform if pathInfo was changed during canonization
+            // redirect: redirect to canonical path
+            // 404: throw NotFound exception
+            // none: no action, just route by canonized path
+            'action' => 'none'
+        ];
```

Comment:
Does not seem right: it should be either set of separater options of the `UrlRule`|`UrlManager` or a seprated class.